src/models/data.py

```diff
@@ -248,13 +368,18 @@ def __next__(self) -> Tuple[Tuple[Union[np.ndarray, torch.Tensor],
                     cur_max_idx = min(cur_max_idx + 1, self.max_idx)
 
                 out_x.append(arrays.x.historical)
+
+                out_x_add.append(arrays.x.current)
```

Comment:
Shouldn't this be a new list?

Otherwise, `out_x_add` will be twice as long as `out_x`

I think the zeroth dimension of the tensors / arrays should always be batch size, which they wouldn't be here

Reply:
can't it be:
```batch_size * 2 ```

so there are two vectors of batch size concatenated to each other?

Reply:
The trouble is the chunking function shuffles based on the first dimension.

So when the x arrays are passed to the chunking function [here](https://github.com/esowc/ml_drought/blob/4f72a41c2c78df0c236873cddb6547dd8c5153a4/src/models/regression.py#L65), I am not sure what the behaviour will be with differently sized first dimensions, but its possible they get screwed up (i.e. data from `x[0]` and `x[1]` are no longer aligned)

Reply:
Plus, it makes it way easier to pass the arrays to the models (see below)

Reply:
good point!

Reply:
Should we not try and merge the current and pred months into one array though?
`(latlons, n_features + 1)` 1 being the pred months vector?

Reply:
Sorry so do you think we should return :

Current behaviour:
```python
return (final_x, final_x_add), final_y
```

What I understand you mean:
```
return (final_x, final_x_add, final_x_curr), final_y
```

The reason I wanted to turn `final_x_add` into an array of shape `(latlons, (n_features - target_var) + 1)` is that the second index (`[1]`) will then be able to be n dimensional if we define other experiments where other vectors are needed

Reply:
Couldn't other experiments just return more arrays in that tuple?

Reply:
very good point